cmd/qri.go

```diff
@@ -92,6 +93,8 @@ type QriOptions struct {
 	repoPath string
 	// generator is source of generating cryptographic info
 	generator gen.CryptoGenerator
+	// automatically run migrations if necessary
```

Comment:
So after testing a few different variations, I think this is the subtly that needs to be defined:

`qri list` -> if there is a migration needed, we will prompt the user for permission to run a migration
`qri list --no-prompt` -> if there is a migration needed, we will inform the user there is a migration needed, but not prompt them to execute one
`qri list --migrate` -> essentially the same as `qri list`, if there is migration we will prompt the user to run one
`qri list --migrate --no-prompt` -> if there is a migration needed, we will begin the migration without prompting the user

So the `migrate` flag really only does anything if the `--no-prompt` flag is also present.

Wondering if we should explain this subtly in the comments and potentially in the flag description => "when used with the `no-prompt` flag, will automatically run migrations if necessary"

Reply:
More worried about future devs understanding the subtleties rather than users. Can see future confusion about "But I have the `--migrate` flag on!" & the description saying that we "automatically run migrations"

Reply:
> qri list --migrate -> essentially the same as qri list, if there is migration we will prompt the user to run one

hrm, that's def not what I intended, and thinking back I didn't properly vet that combination. I think running with the `--migrate` flag should:

* skip the prompt
* run the migration
* continue with the command, not asking the user to re-run.

The last step is up for debate, but I think it honors the "automatic" term in the `migrate` flag description.

Reply:
Yea, what you've outlined as your intention for the command feels correct for the "automatic" term.